### PR TITLE
Feat/restore rest products

### PR DESCRIPTION
**Pull request by ffiebig**
Also implemented Oneclick Mall Deferred capture the best I could given the time we have.

### CODE REVIEW

File: lib/transbank/sdk/oneclick/oneclick.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
 require 'transbank/sdk/oneclick/errors/oneclick_error'
 require 'transbank/sdk/oneclick/inscription/inscription'
-require 'transbank/sdk/oneclick/mall/oneclick_mall'
\ No newline at end of file
+require 'transbank/sdk/oneclick/mall/oneclick_mall'
+require 'transbank/sdk/oneclick/mall_deferred/mall_deferred_transaction'
+require 'transbank/sdk/oneclick/mall_deferred/mall_deferred_inscription'
```
Comment (afiebig) [inline lib/transbank/sdk/oneclick/oneclick.rb:5]:
```suggestion
require 'transbank/sdk/oneclick/mall_deferred/mall_deferred_inscription'

```

File: lib/transbank/sdk/patpass/patpass_by_webpay/base.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          def configure_for_testing
+            @api_key = DEFAULT_API_KEY
+            @commerce_code = DEFAULT_COMMERCE_CODE
+            self.integration_type = :TEST
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_by_webpay/base.rb:47]:
```suggestion
end

```

File: lib/transbank/sdk/patpass/patpass_by_webpay/errors/patpass_by_webpay_errors.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+require 'transbank/sdk/patpass/patpass_by_webpay/errors/transaction_status_error'
+
+module Transbank
+  module Patpass
+    module PatpassByWebpay
+      module Errors
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_by_webpay/errors/patpass_by_webpay_errors.rb:12]:
```suggestion
end

```

File: lib/transbank/sdk/patpass/patpass_by_webpay/errors/transaction_create_error.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+          attr_reader :code
+          def initialize(message, code)
+            @code = code
+            super(message, code)
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_by_webpay/errors/transaction_create_error.rb:15]:
```suggestion
end

```

File: lib/transbank/sdk/patpass/patpass_by_webpay/errors/transaction_status_error.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+          attr_reader :code
+          def initialize(message, code)
+            @code = code
+            super(message, code)
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_by_webpay/errors/transaction_status_error.rb:15]:
```suggestion
end

```

File: lib/transbank/sdk/patpass/patpass_by_webpay/options.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+module Transbank
+  module Patpass
+    module PatpassByWebpay
+      class Options
+        attr_accessor :api_key, :commerce_code, :integration_type
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_by_webpay/options.rb:9]:
```suggestion
end

```

File: lib/transbank/sdk/patpass/patpass_by_webpay/patpass_by_webpay.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+require 'transbank/sdk/patpass/patpass_by_webpay/options'
+require 'transbank/sdk/patpass/patpass_by_webpay/transaction'
+
+
+module Transbank
+  module Patpass
+    module PatpassByWebpay
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_by_webpay/patpass_by_webpay.rb:16]:
```suggestion
end

```

File: lib/transbank/sdk/patpass/patpass_by_webpay/responses/transaction_commit_response.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+        attr_accessor *FIELDS
+
+        def initialize(json)
+          FIELDS.each {|field| send("#{field}=", json["#{field}"]) }
+        end
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_by_webpay/responses/transaction_commit_response.rb:20]:
```suggestion
end

```

File: lib/transbank/sdk/patpass/patpass_comercio/base.rb [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+          def configure_for_testing
+            @api_key = DEFAULT_API_KEY
+            @commerce_code = DEFAULT_COMMERCE_CODE
+            self.integration_type = :TEST
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_comercio/base.rb:47]:
```suggestion
end

```

File: lib/transbank/sdk/patpass/patpass_comercio/errors/patpass_comercio_errors.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+require 'transbank/sdk/patpass/patpass_comercio/errors/inscription_start_error'
+require 'transbank/sdk/patpass/patpass_comercio/errors/inscription_status_error'
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_comercio/errors/patpass_comercio_errors.rb:2]:
```suggestion
require 'transbank/sdk/patpass/patpass_comercio/errors/inscription_status_error'

```

File: lib/transbank/sdk/patpass/patpass_comercio/inscription.rb [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+                commerce_code: Patpass::PatpassComercio::Base::DEFAULT_COMMERCE_CODE,
+                integration_type: Patpass::PatpassComercio::Base::integration_type,
+                base_url: Patpass::PatpassComercio::Base::current_integration_type_url
+            }
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_comercio/inscription.rb:79]:
```suggestion
end

```

File: lib/transbank/sdk/patpass/patpass_comercio/options.rb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+module Transbank
+  module Patpass
+    module PatpassComercio
+      class Options
+        attr_accessor :api_key, :commerce_code, :integration_type
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_comercio/options.rb:9]:
```suggestion
end

```

File: lib/transbank/sdk/patpass/patpass_comercio/responses/patpass_comercio_responses.rb [hunk 13]
```diff
@@ -0,0 +1,2 @@
+require 'transbank/sdk/patpass/patpass_comercio/responses/inscription_start_response'
+require 'transbank/sdk/patpass/patpass_comercio/responses/inscription_status_response'
```
Comment (afiebig) [inline lib/transbank/sdk/patpass/patpass_comercio/responses/patpass_comercio_responses.rb:2]:
```suggestion
require 'transbank/sdk/patpass/patpass_comercio/responses/inscription_status_response'

```

File: lib/transbank/sdk/transaccion_completa/mall/responses/mall_transaction_commit_response.rb [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+                  buy_order session_id card_detail expiration_date
+                  accounting_date transaction_date details
+                  )
+      attr_accessor *FIELDS
+      def initialize(json)
+        FIELDS.each { |field| send("#{field}=", json["#{field}"])}
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/transaccion_completa/mall/responses/mall_transaction_commit_response.rb:14]:
```suggestion
end

```

File: lib/transbank/sdk/transaccion_completa/mall/responses/mall_transaction_installments_response.rb [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+  module TransaccionCompleta
+    class MallTransactionInstallmentsResponse
+
+      attr_accessor :value
+      def initialize(responses)
+        @value = responses
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/transaccion_completa/mall/responses/mall_transaction_installments_response.rb:11]:
```suggestion
end

```

File: lib/transbank/sdk/transaccion_completa/mall/responses/mall_transaction_responses.rb [hunk 16]
```diff
@@ -0,0 +1,3 @@
+require 'transbank/sdk/transaccion_completa/mall/responses/mall_transaction_commit_response'
+require 'transbank/sdk/transaccion_completa/mall/responses/mall_transaction_status_response'
+require 'transbank/sdk/transaccion_completa/mall/responses/mall_transaction_installments_response'
```
Comment (afiebig) [inline lib/transbank/sdk/transaccion_completa/mall/responses/mall_transaction_responses.rb:3]:
```suggestion
require 'transbank/sdk/transaccion_completa/mall/responses/mall_transaction_installments_response'

```

File: lib/transbank/sdk/transaccion_completa/mall/responses/mall_transaction_status_response.rb [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+                  buy_order session_id card_detail expiration_date
+                  accounting_date transaction_date details
+                  )
+      attr_accessor *FIELDS
+      def initialize(json)
+        FIELDS.each { |field| send("#{field}=", json["#{field}"])}
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/transaccion_completa/mall/responses/mall_transaction_status_response.rb:14]:
```suggestion
end

```

File: lib/transbank/sdk/webpay/webpay_plus/base.rb [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+            @api_key = DEFAULT_API_KEY
+            @commerce_code = DEFAULT_MALL_COMMERCE_CODE
+            self.integration_type = :TEST
+          end
+
+        end
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/webpay/webpay_plus/base.rb:63]:
```suggestion
end

```

File: lib/transbank/sdk/webpay/webpay_plus/deferred/responses/webpay_plus_deferred_responses.rb [hunk 19]
```diff
@@ -0,0 +1 @@
+require 'transbank/sdk/webpay/webpay_plus/deferred/responses/transaction_capture_response'
```
Comment (afiebig) [inline lib/transbank/sdk/webpay/webpay_plus/deferred/responses/webpay_plus_deferred_responses.rb:1]:
```suggestion
require 'transbank/sdk/webpay/webpay_plus/deferred/responses/transaction_capture_response'

```

File: lib/transbank/sdk/webpay/webpay_plus/mall/mall_transaction.rb [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+              integration_type: WebpayPlus::Base::integration_type,
+              base_url: WebpayPlus::Base::current_integration_type_url
+            }
+          end
+
+        end
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/webpay/webpay_plus/mall/mall_transaction.rb:89]:
```suggestion
end

```

File: lib/transbank/sdk/webpay/webpay_plus/mall/responses/webpay_plus_mall_responses.rb [hunk 21]
```diff
@@ -0,0 +1,2 @@
+require 'transbank/sdk/webpay/webpay_plus/mall/responses/mall_transaction_status_response'
+require 'transbank/sdk/webpay/webpay_plus/mall/responses/mall_transaction_commit_response'
```
Comment (afiebig) [inline lib/transbank/sdk/webpay/webpay_plus/mall/responses/webpay_plus_mall_responses.rb:2]:
```suggestion
require 'transbank/sdk/webpay/webpay_plus/mall/responses/mall_transaction_commit_response'

```

File: lib/transbank/sdk/webpay/webpay_plus/mall_deferred/mall_deferred_transaction.rb [hunk 22]
```diff
@@ -0,0 +1,110 @@
...
+              integration_type: WebpayPlus::Base::integration_type,
+              base_url: WebpayPlus::Base::current_integration_type_url
+            }
+          end
+
+        end
+      end
+    end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/webpay/webpay_plus/mall_deferred/mall_deferred_transaction.rb:110]:
```suggestion
end

```

File: lib/transbank/sdk/webpay/webpay_plus/options.rb [hunk 23]
```diff
@@ -0,0 +1,11 @@
+module Transbank
+  module Webpay
+    module WebpayPlus
+      class Options
+        attr_accessor :api_key, :commerce_code, :integration_type
+      end
+    end
+  end
+end
+
```
Comment (afiebig) [inline lib/transbank/sdk/webpay/webpay_plus/options.rb:10]:
```suggestion
```

File: lib/transbank/sdk/webpay/webpay_plus/webpay_plus.rb [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+
+require 'transbank/sdk/webpay/webpay_plus/mall_deferred/mall_deferred_transaction'
+
+
+module Transbank
+  module Webpay
+     module WebpayPlus
+     end
+  end
+end
```
Comment (afiebig) [inline lib/transbank/sdk/webpay/webpay_plus/webpay_plus.rb:24]:
```suggestion
end

```

**Review by gonzunigad — APPROVED** [review]

**Comment by gonzunigad** [comment]
Later we can refactor and clean some repeated code. 👍

**Comment by ffiebig** [comment]
> Later we can refactor and clean some repeated code. 👍

Yes please !!! It hurt my heart having to repeat all that code all over again just because we don't have time 😢

**Pull request merged**